src/py_pecg/Subfunction_Module.py

```diff
@@ -0,0 +1,332 @@
+import numpy as np
+
+
+def buscamin(x: np.ndarray) -> int:
+    """
+    This function searches the first local minimum of the modulus of an array x,
+    While truncating the edges (First and Final Indices).
+
+    Args:
+        x (np.ndarray): The target array.
+
+    Returns:
+        ind (int): The first index in x for which the first minimum is located.
+
+    Credits:
+        Original authors: Juan Pablo Martínez, Rute Almeida, Salvador Olmos,
+        Ana Paula Rocha, Pablo Laguna
+        Original Publication: A wavelet-based ECG delineator:
+        evaluation on standard databases
+        Original DOI: 10.1109/TBME.2003.821031.
+    """
+    x = np.abs(x)
+    localmin = (x[1:-1] <= x[:-2]) & (x[1:-1] <= x[2:])
+    ind = np.argmax(localmin)
+    return ind
+
+
+def searchon(piconset: int, sig: np.ndarray, K: float) -> int:
+    """
+    This function searches the onset of a wave using the derivative method.
+
+    Args:
+        piconset (int): Position of the first relevant onset in the wavelet.
+        sig (np.ndarray): Wavelet Signal (in a single scale).
+        K (float): Threshold Factor.
+
+    Returns:
+        onset (int): Resultant index of the wave onset.
+
+    Credits:
+        Original authors: Juan Pablo Martínez, Rute Almeida, Salvador Olmos,
+        Ana Paula Rocha, Pablo Laguna
+        Original Publication: A wavelet-based ECG delineator:
+        evaluation on standard databases
+        Original DOI: 10.1109/TBME.2003.821031.
+    """
+    if K == 0:
+        raise ZeroDivisionError("K cannot be zero!")
+
+    if sig.ndim > 1 and sig.shape[0] > 1 and sig.shape[1] > 1:
+        raise ValueError(
+            "Input is a matrix or higher-dimensional array, not a row or column vector."
+        )
+    elif sig.ndim > 2:
+        raise ValueError("Input is a higher-dimensional array, not a vector.")
```

Comment:
We can support any array which is truly single dimensional, no matter how many dimensions it has in the NumPy array representation.

You can check it in one condition using NumPy `squeeze()`:
```suggestion
   if sig.squeeze().ndim > 1:
        raise ValueError("Input is a higher-dimensional array, not a vector.")
```

And I'd `squeeze()` the signal before using it, to make the the dimensions of the array more predictable for the next lines:
```
sig = sig.squeeze()
```